src/pybind/mgr/dashboard/controllers/role.py

```diff
@@ -1,15 +1,110 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
 
-from . import ApiController, RESTController
-from ..security import Scope
+import cherrypy
+
+from . import ApiController, RESTController, UiApiController
+from ..exceptions import RoleDoesNotExist, DashboardException,\
+    RoleIsAssociatedWithUser, RoleAlreadyExists
+from ..security import Scope as SecurityScope, Permission
 from ..services.access_control import ACCESS_CTRL_DB, SYSTEM_ROLES
 
 
-@ApiController('/role', Scope.USER)
+@ApiController('/role', SecurityScope.USER)
 class Role(RESTController):
+    @staticmethod
+    def _role_to_dict(role):
+        role_dict = role.to_dict()
+        role_dict['system'] = role_dict['name'] in SYSTEM_ROLES
+        return role_dict
+
+    @staticmethod
+    def _validate_permissions(scopes_permissions):
+        if scopes_permissions:
+            for scope in scopes_permissions.keys():
```

Comment:
you could write the loop as `for scope, permissions in scopes_permissions.items():` and drop line 29